app/javascript/packs/application.jsx

```diff
@@ -1,31 +1,74 @@
-/* eslint no-console:0 */
-// This file is automatically compiled by Webpack, along with any other files
-// present in this directory. You're encouraged to place your actual application logic in
-// a relevant structure within app/javascript and only use these pack files to reference
-// that code so it'll be compiled.
-//
-// To reference this file, add <%= javascript_pack_tag 'application' %> to the appropriate
-// layout file, like app/views/layouts/application.html.erb
-
-
-// Uncomment to copy all static images under ../images to the output folder and reference
-// them with the image_pack_tag helper in views (e.g <%= image_pack_tag 'rails.png' %>)
-// or the `imagePath` JavaScript helper below.
-//
-// const images = require.context('../images', true)
-// const imagePath = (name) => images(name, true)
-
-console.log('Hello World from Webpacker')
-
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from '../components/App';
 import history from './history';
-import {Router, Route} from 'react-router-dom';
+import SignUp from "../components/SignUp";
+import SignIn from "../components/SignIn";
+import SignOut from "../components/SignOut";
+import {Router, Route, NavLink, Switch} from 'react-router-dom';
+
+const RegisterNavigation = () => (
+  <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
+    <ul className="navbar-nav mr-auto">
+      <li className="nav-item">
+        <NavLink exact className="nav-link" activeClassName="active" to="/sign_up">Sign up</NavLink>
+      </li>
+      <li className="nav-item">
+        <NavLink exact className="nav-link" activeClassName="active" to="/sign_in">Sign in</NavLink>
+      </li>
+    </ul>
+  </nav>
+);
+
+const SignOutNavigation = () => (
+  <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
+    <ul className="navbar-nav mr-auto">
+      <li className="nav-item">
+        <NavLink exact className="nav-link" activeClassName="active" to="/sign_out">Sign out</NavLink>
+      </li>
+    </ul>
+  </nav>
+);
+
+const MainNavigation = () => (
+  <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
+    <ul className="navbar-nav mr-auto">
+      <li className="nav-item">
+        <NavLink exact className="nav-link" activeClassName="active" to="/">Main</NavLink>
+      </li>
+    </ul>
+  </nav>
+);
+
+const DefaultLayout = ({component: Component, ...rest}) => {
+  let navbarComponent;
+  if (sessionStorage.user == null) {
+    navbarComponent = <RegisterNavigation/>
+  } else {
+    navbarComponent = <SignOutNavigation/>
+  }
+  return (
+    <Route {...rest} render={matchProps => (
+      <div className="DefaultLayout">
+        <div className="Header">{navbarComponent}<MainNavigation/></div>
+        <Component {...matchProps} />
+      </div>
+    )} />
+  )
+};
+
+const Main = () => (
+  <Switch>
+    <DefaultLayout exact path="/" component={App} />
+    <DefaultLayout path="/sign_up" component={ SignUp }/>
```

Comment:
Please remove spaces in curly braces
Also please take a look at React styleguide https://github.com/airbnb/javascript/tree/master/react

Reply:
as for the React styleguide, maybe we’ll wait for the linter? I do not see any obvious errors right now

Reply:
I was talking just about spaces inside brackets and provided link to learn other rules